tests/system_tests/test_artifacts/test_registry.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+import pytest
+import wandb
+
+
+@pytest.mark.parametrize(
+    (
+        "link_artifact_path",
+        "expected_entity",
+        "expected_org",
+        "expected_project",
+        "expected_collection",
+    ),
+    (
+        (
+            "org-name/wandb-registry-model/test-collection",
+            "",
+            "org-name",
+            "wandb-registry-model",
+            "test-collection",
+        ),
+        (
+            "org-entity-name/wandb-registry-model/test-collection",
+            "",
+            "org-entity-name",
+            "wandb-registry-model",
+            "test-collection",
+        ),
+        (
+            "wandb-registry-model/test-collection",
+            "",
+            "",
+            "wandb-registry-model",
+            "test-collection",
+        ),
+        (
+            "random-entity/not-registry/test-collection",
+            "random-entity",
+            "",
+            "not-registry",
+            "test-collection",
+        ),
+        ("not-registry/test-collection", "", "", "not-registry", "test-collection"),
+    ),
+)
+def test_link_artifact_client_handles_registry_paths(
+    tmp_path,
+    user,
+    wandb_init,
+    api,
+    link_artifact_path,
+    expected_entity,
+    expected_org,
+    expected_collection,
+    expected_project,
+    mocker,
+):
```

Comment:
suggestion(if-minor): five variables seems like a lot parameterize simultaneously.  For clarity/readability/maintainability, how would you feel about using dataclasses or namedtuples (instead of plain tuples)?

suggestion below is just meant to be illustrative, feel free to adapt -- or decline -- depending on what you'd prefer:

```python3

@dataclass
class LinkArtifactExpectation:
    entity: str = ""
    org: str = ""
    project: str
    collection: str


@pytest.mark.parametrize(
    (
        "link_artifact_path",
        "expected",
    ),
    (
        (
            "org-name/wandb-registry-model/test-collection",
            LinkArtifactExpectation(
                org="org_name",
                project="wandb-registry-model",
                collection="test-collection",
            ),
        ),
        (
            "org-entity-name/wandb-registry-model/test-collection",
            LinkArtifactExpectation(
                org="org-entity-name",
                project="wandb-registry-model",
                collection="test-collection",
            ),
        ),
        (
            "wandb-registry-model/test-collection",
            LinkArtifactExpectation(
                project="wandb-registry-model",
                collection="test-collection",
            ),
        ),
        (
            "random-entity/not-registry/test-collection",
            LinkArtifactExpectation(
                entity="random-entity",
                project="not-registry",
                collection="test-collection",
            ),
        ),
        (
            "not-registry/test-collection",
            LinkArtifactExpectation(
                project="not-registry", 
                collection="test-collection",
            ),
        ),
    ),
)
def test_link_artifact_client_handles_registry_paths(
    tmp_path,
    user,
    wandb_init,
    api,
    link_artifact_path,
    expected,
    mocker,
):
    expected_entity = expected.entity
    expected_org = expected.org
    expected_collection = expected.collection
    expected_project = expected.project
```

Reply:
yup that looks alot better